internal/payout/txdetails.go

```diff
@@ -0,0 +1,50 @@
+package payout
+
+import (
+	"github.com/NodeFactoryIo/go-substrate-rpc-client/rpc/author"
+	log "github.com/sirupsen/logrus"
+	"math/big"
+)
+
+type TransactionStatus string
+
+const (
+	Finalized = TransactionStatus("Finalized")
+	Dropped   = TransactionStatus("Dropped")
+	Invalid   = TransactionStatus("Invalid")
+)
+
+type TransactionDetails struct {
+	To     string
+	Amount big.Int
+	Status TransactionStatus
+}
+
+func listenForTransactionStatus(
```

Comment:
Can you write tests for this?

Reply:
Actually, this is not trivial as the first argument `sub *author.ExtrinsicStatusSubscription` doesn't define an interface for its methods so I should implement some kind of `IntermidiateLayer` interface (then I could mock its methods).

I actually wanted to do this for `api *gsrpc.SubstrateAPI` and then use this mock to test this entire flow but I moved this to a separate issue #122.